exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/StreamingAggBatch.java

```diff
@@ -298,59 +305,63 @@ public IterOutcome innerNext() {
           // If outcome is NONE then we send the special batch in the first iteration and the NONE
           // outcome in the next iteration. If outcome is EMIT, we can send the special
           // batch and the EMIT outcome at the same time.
-
-          IterOutcome finalOutcome =  getFinalOutcome();
-          return finalOutcome;
+          return getFinalOutcome();
         }
         firstBatchForDataSet = true;
         firstBatchForSchema = false;
         if(first) {
           first = false;
         }
-        if(lastKnownOutcome == OK_NEW_SCHEMA) {
-          sendEmit = true;
+        if(returnOutcome == OK_NEW_SCHEMA) {
+          sendEmit = (aggregator == null) || aggregator.previousBatchProcessed();
```

Comment:
`aggregator` should ideally not be null here. This is just to avoid NPE and since returned AggOutcome is `RETURN_AND_RESET` with `OK_NEW_SCHEMA` then the next batch should be an empty batch with `EMIT` outcome.